src/blockdev.rs

```diff
@@ -91,27 +93,13 @@ fn get_partitions(device: &str) -> Result<Vec<BlkDev>> {
     for line in output.lines() {
         // parse key-value pairs
         let fields = split_lsblk_line(line);
-
-        // Older lsblk, e.g. in CentOS 7.6, doesn't support PATH.
-        // Assemble device path from dirname and NAME.
-        let mut path = Path::new(device)
-            .parent()
-            .chain_err(|| format!("couldn't get parent directory of {}", device))?
-            .to_path_buf();
-        match fields.get("NAME") {
-            None => continue,
-            Some(name) => path.push(name),
-        }
-        // Skip the device itself
-        if path == Path::new(device) {
-            continue;
```

Comment:
I think we still need some logic like this though, but I guess comparing to `/dev/${device}` instead?

Reply:
I guess `device` always contains `/dev/` prefix.
But i don't think we really need to filter out device itself, the only reason to do it is to fail with `couldn't find any partitions on {device}`, i think it's better to fail with `couldn't find {} device for {label}`